android/src/main/java/com/baseflow/googleapiavailability/GoogleApiAvailabilityManager.java

```diff
@@ -0,0 +1,147 @@
+package com.baseflow.googleapiavailability;
+
+import android.app.Activity;
+import android.app.Dialog;
+import android.app.PendingIntent;
+import android.content.Context;
+import android.util.Log;
+
+import com.google.android.gms.common.ConnectionResult;
+import com.google.android.gms.common.GoogleApiAvailability;
+
+import java.util.List;
+
+public class GoogleApiAvailabilityManager {
+
+    GoogleApiAvailability googleApiAvailability = GoogleApiAvailability.getInstance();
+
+    @FunctionalInterface
+    interface SuccessCallback {
+        void onSuccess(int connectionResult);
+    }
+
+    @FunctionalInterface
+    interface MakeGooglePlayServicesAvailableCallback {
+        void onSuccess(boolean makeGooglePlayServicesAvailable);
```

Comment:
Consider updating this callback to not except a parameter and use the `errorCallback` to signal a failure (see also feedback on the `makeGooglePlayServicesAvailable` method:

```suggestion
        void onSuccess();
```